pkg/network/node/ovscontroller.go

```diff
@@ -283,33 +283,56 @@ func (oc *ovsController) SetUpPod(hostVeth, podIP, podMAC, sandboxID string, vni
 	if err != nil {
 		return -1, err
 	}
-	ofport, err := oc.ensureOvsPort(hostVeth)
+	ofport, err := oc.ensureOvsPort(hostVeth, sandboxID)
 	if err != nil {
 		return -1, err
 	}
 	return ofport, oc.setupPodFlows(ofport, podIP, podMAC, note, vnid)
 }
 
-func (oc *ovsController) SetPodBandwidth(hostVeth string, ingressBPS, egressBPS int64) error {
-	// note pod ingress == OVS egress and vice versa
+func (oc *ovsController) getPortsForSandbox(sandboxID string) ([]string, error) {
+	return oc.ovs.Find("port", "external-ids=sandbox="+sandboxID)
+}
 
-	qos, err := oc.ovs.Get("port", hostVeth, "qos")
-	if err != nil {
-		return err
-	}
-	if qos != "[]" {
-		err = oc.ovs.Clear("port", hostVeth, "qos")
+func (oc *ovsController) ClearPodBandwidth(portList []string, sandboxID string) error {
+	var err error
+
+	// Clear the QoS for any ports of this sandbox
+	if len(portList) == 0 {
+		portList, err = oc.getPortsForSandbox(sandboxID)
 		if err != nil {
 			return err
 		}
-		err = oc.ovs.Destroy("qos", qos)
-		if err != nil {
+	}
+	for _, port := range portList {
+		if err := oc.ovs.Clear("port", port, "qos"); err != nil {
 			return err
 		}
 	}
 
+	// Now that the QoS is unused remove it
+	qosList, err := oc.ovs.Find("qos", "external-ids=sandbox="+sandboxID)
+	if err != nil {
+		return err
+	}
+	for _, qos := range qosList {
+		if err := oc.ovs.Destroy("qos", qos); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (oc *ovsController) SetPodBandwidth(hostVeth, sandboxID string, ingressBPS, egressBPS int64) error {
+	// note pod ingress == OVS egress and vice versa
+
+	if err := oc.ClearPodBandwidth(nil, sandboxID); err != nil {
```

Comment:
You could remove the `hostVeth` arg to SetPodBandwidth() and just call getPortsForSandbox() and then pass that list to ClearPodBandwidth() (which then no longer needs its own call-getPortsForSandbox() special case), and change the remaining hostVeth references in SetPodBandwidth() to use the portsList instead. (I'm not totally sure if this would be a net improvement.)

Reply:
@danwinship I'd rather still pass it in so we know what the most-recently-created veth will be for actually setting the bandwidth.  Otherwise we'd have to assume it's the first item in the ports array, or do it for all ports.  We already have the info in the veth name here.

But I agree we don't need ClearPodBandwidth() to do its own getPortsForSandboxID().